provisioning/device/src/client.ts

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+'use strict';
+import * as Provisioning from './interfaces';
+import { ClientStateMachine } from './client_state_machine';
+import * as dbg from 'debug';
+const debug = dbg('azure-device-provisioning:device-client');
+
+export class ProvisioningDeviceClient implements Provisioning.ClientConfiguration {
```

Comment:
>ClientConfiguration [](start = 62, length = 19)

single responsibility principle question here: it's both the client and the config object? feels like those two should be separate.

Reply:
agree.  refactoring in later commit.

---
In reply to: [148918192](https://github.com/Azure/azure-iot-sdk-node/pull/135#discussion_r148918192) [](ancestors = 148918192)